### PR TITLE
Refactor Airlock Modules

### PR DESCRIPTION
**Pull request by yuvalyaron**
# Preliminary to #2829

## What is being addressed

- Moved the logic of `create_review_user_resource` from the airlock route to the airlock service, and broke it into multiple shorter methods
- Removed the `airlock_resource_helpers` module, moved its methods to the airlock service.
- Aligned tests with the changes - no changes were made to the logic of the tests

### CODE REVIEW

File: api_app/api/routes/airlock_resource_helpers.py [hunk 1]
```diff
@@ -1,204 +0,0 @@
-from datetime import datetime
```
Comment (yuvalyaron) [inline api_app/api/routes/airlock_resource_helpers.py:1]: all the methods here were moved to the airlock service

File: api_app/services/airlock.py [hunk 2]
```diff
@@ -91,3 +114,319 @@ def get_airlock_request_container_sas_token(account_name: str,
...
+        _ = await delete_review_user_resource(
+            user_resource=existing_resource,
+            user_resource_repo=user_resource_repo,
+            workspace_service_repo=workspace_service_repo,
+            resource_template_repo=resource_template_repo,
+            operations_repo=operation_repo,
+            resource_history_repo=resource_history_repo,
+            user=user
+        )
+
```
Comment (yuvalyaron) [inline api_app/services/airlock.py:254]: this is the logic from the airlock route for `create_review_user_resource`,
broke it to multiple methods

File: api_app/tests_ma/conftest.py [hunk 3]
```diff
@@ -458,3 +460,13 @@ def simple_pipeline_step() -> PipelineStep:
             ),
         ]
     )
+
+
+@pytest_asyncio.fixture()
+def no_database():
```
Comment (anatbal) [inline api_app/tests_ma/conftest.py:466]: Why do we not just mock the airlock repo obj?

**Review by anatbal — APPROVED** [review]
Looks good! approved with a comment.
Pls run airlock tests on that since extended don't contain the e2e tests for review vm

File: api_app/tests_ma/conftest.py (hunk 3 above)
Comment (anatbal) [reply api_app/tests_ma/conftest.py:466]: Cool, got it

File: e2e_tests/resources/deployment.py [hunk 4]
```diff
@@ -32,9 +32,20 @@ async def check_deployment(client, operation_endpoint, headers):
...
-        message = response.json()["operation"]["message"]
+        message = parse_message(response)
         return deployment_status, message
     else:
         LOGGER.error(f"Non 200 response in check_deployment: {response.status_code}")
         LOGGER.error(f"Full response: {response}")
         raise Exception("Non 200 response in check_deployment")
+
+
+def parse_message(response):
```
Comment (tamirkamara) [inline e2e_tests/resources/deployment.py:43]: Maybe a better name? something like format_deployment_status_message?

File: e2e_tests/resources/deployment.py [hunk 5]
```diff
@@ -32,9 +32,20 @@ async def check_deployment(client, operation_endpoint, headers):
     response = await client.get(full_endpoint, headers=headers, timeout=TIMEOUT)
     if response.status_code == 200:
         deployment_status = response.json()["operation"]["status"]
```
Comment (tamirkamara) [inline e2e_tests/resources/deployment.py:34]: Now there're 2 response.parse, maybe you should consolidate?

File: e2e_tests/resources/deployment.py (hunk 4 above)
Comment (yuvalyaron) [reply e2e_tests/resources/deployment.py:43]: changed

File: e2e_tests/resources/deployment.py (hunk 5 above)
Comment (yuvalyaron) [reply e2e_tests/resources/deployment.py:34]: done

**Review by tamirkamara — APPROVED** [review]

**Comment by yuvalyaron** [comment]
tested at:
https://github.com/microsoft/AzureTRE/actions/runs/4013913729/jobs/6896913729
test_airlock_review_vm_flow passed
test_airlock_flow is not related and failed in teardown on a known issue

**Pull request merged**